libs/src/services/discoveryProvider/DiscoveryProviderSelection.test.js

```diff
@@ -0,0 +1,363 @@
+        service: 'discovery-provider',
+        version: '1.2.2',
+        block_difference: 0
+      })
+
+    const s = new DiscoveryProviderSelection(
+      {},
+      mockEthContracts([healthy, outdated], '1.2.3')
+    )
+    const service = await s.select()
+    assert.strictEqual(service, healthy)
+  })
+
+  it('prefers a healthy block diff', async () => {
+    const healthy = 'https://healthy.audius.co'
+    nock(healthy)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 0
+      })
+    const behind = 'https://behind.audius.co'
+    nock(behind)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 20
+      })
+
+    const s = new DiscoveryProviderSelection(
+      {},
+      mockEthContracts([healthy, behind], '1.2.3')
+    )
+    const service = await s.select()
+    assert.strictEqual(service, healthy)
+  })
+
+  it('can select an old version', async () => {
+    const healthyButBehind = 'https://healthyButBehind.audius.co'
+    nock(healthyButBehind)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 20
+      })
+    const pastVersionNotBehind = 'https://pastVersionNotBehind.audius.co'
+    nock(pastVersionNotBehind)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.2',
+        block_difference: 0
+      })
+
+    const s = new DiscoveryProviderSelection(
+      { requestTimeout: 100 },
+      mockEthContracts([healthyButBehind, pastVersionNotBehind], '1.2.3')
+    )
+    const service = await s.select()
+    assert.strictEqual(service, pastVersionNotBehind)
+    assert.deepStrictEqual(s.backups, {
+      [healthyButBehind]: {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 20
+      },
+      [pastVersionNotBehind]: {
+        service: 'discovery-provider',
+        version: '1.2.2',
+        block_difference: 0
+      }
+    })
+    assert.strictEqual(s.getTotalAttempts(), 2)
+  })
+
+  it('can select the discprov that is the least number of blocks behind', async () => {
+    const behind20 = 'https://behind20.audius.co'
+    nock(behind20)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.2',
+        block_difference: 20
+      })
+    const behind40 = 'https://behind40.audius.co'
+    nock(behind40)
+      .get('/health_check')
+      .reply(200, {
+        service: 'discovery-provider',
+        version: '1.2.3',
+        block_difference: 40
+      })
+
+    const s = new DiscoveryProviderSelection(
+      { requestTimeout: 100 },
+      mockEthContracts([behind20, behind40], '1.2.3')
+    )
+    const service = await s.select()
+    assert.strictEqual(service, behind40)
```

Comment:
looks like this is not on par with the test description of picking the dp that has the least number of blocks behind? or is the description supposed to describe how `behind20` can be added as a backup?

Reply:
good catch! added more to the test name!
`can select the discprov that is the least number of blocks behind for the current version`